packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,428 @@
+        name={this.props.granularitySelectName}
+        defaultValue={this.state.granularity}
+        onChange={this.changeGranularity}
+        isInline
+      />);
+    }
+
+    let ageSelect = null;
+    let ageUnitSelect = null;
+    if (this.state.granularity === 'AGE' && this.state.precision !== 'UNKNOWN') {
+      ageSelect = (<NumberField
+        className="terra-OnsetPicker-age"
+        min={1}
+        max={this.maxAgeCount()}
+        step={1}
+        value={this.state.age.toString()}
+        onChange={this.changeAge}
+        isInline
+      />);
+
+      ageUnitSelect = (<SelectField
+        className="terra-OnsetPicker-ageUnit"
+        options={this.allowedAgeDurations()}
+        defaultValue={this.state.ageUnit.toString()}
+        onChange={this.changeAgeUnit}
+        isInline
+      />);
+    }
+
+    let monthSelect = null;
+    if (this.state.granularity === 'MONTH' && this.state.precision !== 'UNKNOWN') {
+      monthSelect = (<SelectField
+        className="terra-OnsetPicker-month"
+        options={this.availableMonths(intl)}
+        defaultValue={moment(this.state.onsetDate).month().toString()}
+        onChange={this.changeMonth}
+        isInline
+      />);
+    }
+
+    let yearSelect = null;
+    if ((this.state.granularity === 'YEAR' || this.state.granularity === 'MONTH') && this.state.precision !== 'UNKNOWN') {
+      yearSelect = (<SelectField
+        className="terra-OnsetPicker-year"
+        options={this.availableYears()}
+        defaultValue={moment(this.state.onsetDate).year().toString()}
+        onChange={this.changeYear}
+        isInline
+      />);
+    }
+
+    let dateSelect = null;
+    if (this.state.granularity === 'DATE' && this.state.precision !== 'UNKNOWN') {
+      dateSelect = (<Field>
+        <DatePicker
+          onChange={this.changeDate}
+          minDate={this.props.birthdate}
+          maxDate={moment().format('YYYY-MM-DD')}
+          selectedDate={moment(this.state.onsetDate).format('YYYY-MM-DD')}
+          name={this.props.onsetDateInputName}
+        />
+      </Field>);
+    }
+
+    return (
+      (<div className="terra-OnsetPicker" {...customProps}>
+
+        <Fieldset>
+          {/* Precision */}
+          <SelectField
+            options={[{ value: 'ABOUT', display: intl.formatMessage({ id: 'Terra.onsetPicker.precisionAbout' }) },
+                      { value: 'BEFORE', display: intl.formatMessage({ id: 'Terra.onsetPicker.precisionBefore' }) },
+                      { value: 'AFTER', display: intl.formatMessage({ id: 'Terra.onsetPicker.precisionAfter' }) },
+                      { value: 'UNKNOWN', display: intl.formatMessage({ id: 'Terra.onsetPicker.precisionUnknown' }) }]}
+            name={this.props.precisionSelectName}
+            defaultValue={this.state.precision}
+            onChange={this.changePrecision}
+            isInline
+          />
+
+          {granularitySelect}
+        </Fieldset>
+
+        <Fieldset>
```

Comment:
Since none of these selects will be rendered if precision is UNKNOWN, should we just not render this second Fieldset at all in that case? Then you also wouldn't need to do that check for all 5 of these selects individually.

Reply:
Good point, updated.